scripts/source-maintenance.sh

```diff
@@ -36,13 +38,81 @@ CheckAndUpdateCopyright=yes
 # * SHA1/etc: Check commits added after the specified git commit.
 UpdateContributorsSince=auto
 
+# --only-changed-since point
+OnlyChangedSince=""
+
 printUsage () {
-    echo "Usage: $0 [option...]"
-    echo "options:"
-    echo "    --keep-going|-k"
-    echo "    --check-and-update-copyright <yes|no>"
-    echo "    --update-contributors-since <never|auto|revision>"
-    echo "    --with-astyle </path/to/astyle/executable>"
+cat <<USAGE_
+Usage: $0 [option...]
+options:
+--check-and-update-copyright <yes|no>      (default: yes)
+--help|-h
+--keep-going|-k                            (default: stop on error)
+--only-changed-since <fork|commit>         (default: apply to all files)
+--update-contributors-since <never|auto|revision> (default: auto)
+--with-astyle </path/to/astyle/executable> (default: astyle-${TargetAstyleVersion} or astyle)
+
+USAGE_
+}
+
+printHelp () {
+
+cat <<HELP_INTRO_
+This script applies Squid mandatory code style guidelines and generates
+various files derived from Squid sources.
+
+If the --only-changed-since argument is supplied, it expects a git commit-id,
+branch name or the special keyword 'fork'.
+The script will try to only examine for formatting changes those files that
+have changed since the specified commit.
+The keyword 'fork' will look for files changed
+since the current branch was forked off 'upstream/master'. Sensible values
+for this argument may include HEAD^, master, origin/master, or the branch
+the current one was forked off.
+This option does not disable some repository-wide file generation and
+repository-wide non-formatting checks/adjustments.
```

Comment:
AFAICT, essentially the same information is relayed in `--only-changed-since` documentation below (where it belongs).

```suggestion
```